applications/StructuralMechanicsApplication/python_scripts/trilinos_structural_mechanics_solver.py

```diff
@@ -31,10 +31,22 @@ def __init__(self, model, custom_settings):
     def GetDefaultParameters(cls):
         this_defaults = KratosMultiphysics.Parameters("""{
             "multi_point_constraints_used": false,
-            "linear_solver_settings" : {
-                "solver_type" : "amesos",
-                "amesos_solver_type" : "Amesos_Klu"
-            }
+            "linear_solver_settings":{
+                "solver_type"                    :"amgcl",
+                "preconditioner_type"            : "amg",
+                "max_iteration"                  : 500,
+                "tolerance"                      : 1e-6,
+                "provide_coordinates"            : false,
+                "smoother_type"                  : "ilut",
+                "krylov_type"                    : "bicgstab_with_gmres_fallback",
```

Comment:
I think the reason for `Unknown parameter M` message is hidden somewhere in `bicgstab_with_gmres_fallback` implementation. It sets the number of restarts M for the GMRES even when the BiCGStab is used. This is from my memory (and the name of the parameter), as I don't know where the `bicgstab_with_gmres_fallback` is actually defined.

Reply:
good hint, I think this is it:
https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/linear_solvers/amgcl_solver.h#L449
can it just be removed?

Reply:
The default value for M is 30, and `gmres_krylov_space_dimension` is set to 100 in the options. So removing it could possible change the behavior.

Reply:
I don't see a problem in the [line 449](https://github.com/KratosMultiphysics/Kratos/blob/dfee8e36847897de15ed54784fc428b315df66a3/kratos/linear_solvers/amgcl_solver.h#L449), as it is always used after the solver type is set to "gmres". The [lines 433-438](https://github.com/KratosMultiphysics/Kratos/blob/dfee8e36847897de15ed54784fc428b315df66a3/kratos/linear_solvers/amgcl_solver.h#L433-L438) above also look OK to me, as the "M" parameter seems to be removed for non-gmres solvers. So I am not sure where the warning is coming from.

Reply:
The only other line where "solver.M" is set, is [line 809](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/linear_solvers/amgcl_solver.h#L809), and it is used along with setting "solver.type" to "bicgstab". Could that be a problem?